tensorflow_datasets/core/download/download_manager.py

```diff
@@ -238,13 +238,17 @@ def _download(self, resource):
     if isinstance(resource, six.string_types):
       resource = resource_lib.Resource(url=resource)
     url = resource.url
+    download_type = ""
+    if kaggle.KaggleFile.is_kaggle_url(url):
+      download_type = url[-1]
+      url = url[:-1]
```

Comment:
Splitting the last character of the url is confusing for people not familiar with the Kaggle API. Could you add comment explaining what this code does, and give an example of url ?

Reply:
okay, so when it finds a kaggle url the code checks if it's a dataset url or not and adds a suffix to it for further reference.
So, if the link looks like: `kaggle://some-competition/competition-name` it would become, `kaggle://some-competition/competition-name0` and if it's a datasets url, it would become `kaggle://some-dataset/author/dataset-name1`

So, while processing, I'm splitting out this suffix (0 or 1), as `download_type` and keep the rest in the url.
After checksum calculation and logging is done, I again attach the suffix and send it back to downloader, where this suffix will be used to choose how to split the link to obtain the necessary information and pass it to `kaggle_cli`